src/client/pkg/config/config.go

```diff
@@ -42,24 +60,67 @@ func Read() (*Config, error) {
 	} else {
 		return nil, fmt.Errorf("fatal: could not read config at %q: %v", p, err)
 	}
+
+	updated := false
+
 	if c.UserID == "" {
-		fmt.Fprintf(os.Stderr, "No UserID present in config. Generating new UserID and "+
-			"updating config at %s\n", p)
+		updated = true
+		fmt.Fprintln(os.Stderr, "No UserID present in config - generating new one.")
+
 		uuid, err := uuid.NewV4()
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("could not generate new user ID: %v", err)
 		}
 		c.UserID = uuid.String()
+	}
+
+	if c.V2 == nil {
+		updated = true
+		fmt.Fprintln(os.Stderr, "Migrating to config v2.")
+
+		c.V2 = &ConfigV2{
+			ActiveContext: "default",
+			Contexts:      map[string]*Context{},
+		}
+
+		if c.V1 != nil {
+			if _, ok := c.V2.Contexts["default"]; ok {
+				return nil, fmt.Errorf("Attempted to migrate to config v2, but there's already a default context")
+			}
+			c.V2.Contexts["default"] = &Context{
+				Source:            ContextSource_CONFIG_V1,
+				PachdAddress:      c.V1.PachdAddress,
+				ServerCAs:         c.V1.ServerCAs,
+				SessionToken:      c.V1.SessionToken,
+				ActiveTransaction: c.V1.ActiveTransaction,
+			}
+
+			c.V1 = nil
+		} else {
+			c.V2.Contexts["default"] = &Context{
+				Source: ContextSource_NONE,
+			}
+		}
+	}
+
+	if updated {
+		fmt.Fprintln(os.Stderr, "Rewriting config at %q.", p)
+
 		if err := c.Write(); err != nil {
-			return nil, err
+			return nil, fmt.Errorf("could not rewrite config at %q: %v", p, err)
 		}
 	}
+
 	return c, nil
 }
 
 // Write writes the configuration in 'c' to this machine's Pachyderm config
 // file.
 func (c *Config) Write() error {
+	if c.V1 != nil {
+		panic("v1 config included, implying a bug")
```

Comment:
Very minor nit, we generally use the string `(this is likely a bug)` in cases like this. I figure we might as well keep it consistent throughout.